File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:         
        d={}
        while nums!=[]:
            if nums[-1] in d:
                d[nums[-1]]+=1
            else:
                d[nums[-1]]=1
            nums.pop()
        # print(d)
        a = list(d.keys())[0]
        for e in d:
            if d.get(a) < d[e]:
                a=e
        return a

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 1;
        int result = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (result == nums[i])
                count++;
            else
                count--;
            if (count == 0) {
                result = nums[i];
                count++;
            }
        }
        return result;
    }
}